compiler/aarch64/codegen/ARM64Instruction.hpp

```diff
@@ -1569,6 +1569,90 @@ class ARM64Trg1Src2Instruction : public ARM64Trg1Src1Instruction
    virtual uint8_t *generateBinaryEncoding();
    };
 
+class ARM64CondTrg1Src2Instruction : public ARM64Trg1Src2Instruction
+   {
+   TR::ARM64ConditionCode _cc;
+
+   public:
+
+   /*
+    * @brief Constructor
+    * @param[in] op : instruction opcode
+    * @param[in] node : node
+    * @param[in] treg : target register
+    * @param[in] s1reg : source register 1
+    * @param[in] s2reg : source register 2
+    * @param[in] cc : branch condition code
+    * @param[in] cg : CodeGenerator
+    */
+   ARM64CondTrg1Src2Instruction( TR::InstOpCode::Mnemonic op,
+                             TR::Node *node,
+                             TR::Register *treg,
+                             TR::Register *s1reg,
+                             TR::Register *s2reg,
+                             TR::ARM64ConditionCode cc, 
+                             TR::CodeGenerator *cg)
+      : ARM64Trg1Src2Instruction(op, node, treg, s1reg, s2reg, cg), _cc(cc)
+      {
+      }
+
+   /*
+    * @brief Constructor
+    * @param[in] op : instruction opcode
+    * @param[in] node : node
+    * @param[in] treg : target register
+    * @param[in] s1reg : source register 1
+    * @param[in] s2reg : source register 2
+    * @param[in] cc : branch condition code
+    * @param[in] precedingInstruction : preceding instruction
+    * @param[in] cg : CodeGenerator
+    */
+   ARM64CondTrg1Src2Instruction( TR::InstOpCode::Mnemonic op,
+                             TR::Node *node,
+                             TR::Register *treg,
+                             TR::Register *s1reg,
+                             TR::Register *s2reg,
+                             TR::ARM64ConditionCode cc, 
+                             TR::Instruction *precedingInstruction, TR::CodeGenerator *cg)
+      : ARM64Trg1Src2Instruction(op, node, treg, s1reg, s2reg, precedingInstruction, cg), _cc(cc)
+      {
+      }
+      
+   /**
+    * @brief Gets instruction kind
+    * @return instruction kind
+    */
+   virtual Kind getKind() { return IsCondTrg1Src2; }
+
+   /**
+    * @brief Gets condition code
+    * @return condition code
+    */
+   TR::ARM64ConditionCode getConditionCode() {return _cc;}
+
+   /**
+    * @brief Sets condition code
+    * @param[in] cc : condition code
+    * @return condition code
+    */
+   TR::ARM64ConditionCode setConditionCode(TR::ARM64ConditionCode cc) {return (_cc = cc);}
+
+   /**
+    * @brief Sets condition code in binary encoding
+    * @param[in] instruction : instruction cursor
+    */
+   void insertConditionCodeField(uint32_t *instruction)
+      {
+      *instruction |= (_cc & 0xf);
```

Comment:
What instruction encoding are you following (which section of the architecture manual)?  This isn't where the condition code should be encoded.  As mentioned earlier, please see 6.2.61.

Reply:
![image](https://user-images.githubusercontent.com/43818645/60668005-548b7800-9e41-11e9-93bd-a0a18fd4aa33.png)
While going to that reference I found this. Do you want me to follow this one, it would be really helpful if you can please clarify this?

Reply:
That looks like the doc, but perhaps you're looking at an older one?  Even in the document you have, search for the `CSEL` instruction.  I'm sure its semantics have not changed.

Here is the link I've followed for the latest AArch64 guide.  It requires registered ARM access, but that should not be a problem for you (check with @aarongraham9 as we had this discussion many months ago).

https://silver.arm.com/download/download.tm?pv=4535355&p=1343131